src/serializer/functions.js

```diff
@@ -138,6 +138,12 @@ export class Functions {
     let additionalFunctionEffects = this._createAdditionalEffects(effects);
     let value = effects[0];
 
+    // for the failing test, `value` will be a `PossiblyNormalCompletion`
+    // we want the value from it instead? How do we best do that?
+    // furthermore, when we get that value, it needs to be the value used
+    // for this additionalFunction during visiting/serialization so the test
+    // passes
+
```

Comment:
@cblappert This is the point that I'm a bit stuck with. If I try `realm.composeWithSavedCompletion(value);`, it gives me back a `ObjectValue` that looks like a ReactElement, so that's good – but it also makes the `realm.generator._entries` and empty array and this during visiting/serialization. If I instead skip over checking the value here (just to find what is causing the problems), I get an invariant later on when checking all the bindings for their values.

If you check out this branch and use the test provided , you should see the same problem.

Reply:
![screen shot 2018-02-15 at 5 00 34 pm](https://user-images.githubusercontent.com/1519870/36269724-cade2014-1271-11e8-9041-37f154df6632.png)

Reply:
Can't seem to get the debugger working properly on either VSCode or Nuclide (Nuclide hangs, VSCode refuses to understand sourcemaps). Is there anything special you did aside from adding `--runInBand` to jest?

Reply:
I don’t bother debugging via Jest anymore, I just setup the same test in its own file and run all the test-react configuration flags on it.

Reply:
I run `fb-www` for `compatability` using the CLI directly with settings passed on command line.